### PR TITLE
PE-638: Uploads skip on file to file name conflicts by default

### PR DESCRIPTION
**Pull request by fedellen**
This PR adds two additional parameters to the `upload-file` command, `--skip` and `--replace`. 

By default, the command will now use the `skip` conflict resolution. This will now ~~cause an error~~ return an empty ArFS Result on any file to file name conflicts during a single file upload. On a bulk upload, it will likewise skip that file and will continue to try the rest of the upload. When `replace` is set as the conflict resolution, the command will upload a revision as the default behavior.

This new `conflictResolution` parameter has been applied to the `ArDrive` class settings. This is so we can share it for other commands in the future; like `move` and `rename`.

### CODE REVIEW

**Review by arielmelendez — DISMISSED** [review]

File: src/CLICommand/parameters_helper.ts [hunk 1]
```diff
@@ -204,6 +206,22 @@ export class ParametersHelper {
...
+			return 'replace';
+		}
+
+		// if (this.getParameterValue(UpsertParameter)) {
+		// 	return 'upsert'
+		// };
+
+		// if (this.getParameterValue(AskParameter)) {
+		// 	return 'ask'
+		// };
```
[the remark this answers is not in the document]
Comment (matibat) [reply src/CLICommand/parameters_helper.ts:220]: Do we have a ticket for it?

**Review by matibat — APPROVED** [review]

File: src/ardrive.ts [hunk 2]
```diff
@@ -499,9 +526,16 @@ export class ArDrive extends ArDriveAnonymous {
 			throw new Error(errorMessage.entityNameExists);
 		}
 
+		const conflictingFileName = filesAndFolderNames.files.find((f) => f.fileName === destFileName);
+
+		if (conflictingFileName && conflictResolution === 'skip') {
+			return emptyArFSResult;
+		}
+		// TODO: Handle this.conflictResolution === 'upsert' and 'ask' PE-638
```
Comment (matibat) [inline src/ardrive.ts]: @arielmelendez @agsuy @fedellen 
Do you guys think we need another story ticket for it? How are we gonna deal with these cupcakes at the process level?

File: src/CLICommand/parameters_helper.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (matibat) [reply src/CLICommand/parameters_helper.ts]: I think we don't yet have one

**Review by matibat — COMMENTED** [review]
It's looking good to me! I'll approve after I test it a little

File: src/ardrive.ts (hunk 2 above)
Comment (matibat) [reply src/ardrive.ts]: I know we talked about it at some standup, but I don't recall :sweat_smile:

**Review by matibat — COMMENTED** [review]
The skip:
```shell
$ yarn ardrive upload-file --skip -F 450fd857-c9e6-4b38-81b6-8ebdbd5a19d3 -l ~/Downloads/LICENSE.txt -w ../17drivesNoAR_key1_pass_testingPASSword.json -P         
✔ Enter drive password: … ***************
{
    "created": [],
    "tips": [],
    "fees": {}
}
```
The replace:
```shell
$ yarn ardrive upload-file --replace -F 450fd857-c9e6-4b38-81b6-8ebdbd5a19d3 -l ~/Downloads/LICENSE.txt -w ../17drivesNoAR_key1_pass_testingPASSword.json -P --dry-run
✔ Enter drive password: … ***************
{
    "created": [
        {
            "type": "file",
            "metadataTxId": "ntJkMyCguvbMjvTDtCWoo6su6DegQVnJyzsoSxPxKZs",
            "dataTxId": "GFFVXZxsm2oFpBiSv1etjL2qDqO6aSSkbNyWo-RS20s",
            "entityId": "912f88c9-8ddb-4b9c-aff1-8cca9ba28fa1",
            "key": "x/rYw18Mq/6Wqq/bxUKhR3Nqeclq8LPtTY2dAcu4yT0"
        }
    ],
    "tips": [
        {
            "txId": "dF_gRM8oUtQ0NbaassK5eTbAcCbRV6GKYZmXU5ZsStw",
            "recipient": {
                "address": "FAxDUPlFfJrLDl6BvUlPw3EJOEEeg6WQbhiWidU7ueY"
            },
            "winston": "10000000"
        }
    ],
    "fees": {
        "GFFVXZxsm2oFpBiSv1etjL2qDqO6aSSkbNyWo-RS20s": 1848298,
        "ntJkMyCguvbMjvTDtCWoo6su6DegQVnJyzsoSxPxKZs": 1411906,
        "dF_gRM8oUtQ0NbaassK5eTbAcCbRV6GKYZmXU5ZsStw": 1340496
    }
}
```

**Review by matibat — APPROVED** [review]

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**